src/python/WMCore/Services/Rucio/Rucio.py

```diff
@@ -356,29 +357,40 @@ def attachDIDs(self, rse, superDID, dids, scope='cms'):
              then it's a container name; if attaching files, then it's a block name)
         :param dids: either a string or a list of data identifiers (can be block or files)
         :param scope: string with the scope name
+        :param chunkSize: number of dids to send to upstream Rucio server
         :return: a boolean to represent whether it succeeded or not
         """
         if not isinstance(dids, list):
             dids = [dids]
         # NOTE: the attaching dids do not create new container within a scope
         # and it is safe to use cms scope for it
-        dids = [{'scope': 'cms', 'name': did} for did in dids]
+        alldids = [{'scope': 'cms', 'name': did} for did in dids]
```

Comment:
As just chatted, we need to sort this dids list to make this block attachment a deterministic process, otherwise we cannot treat the duplicate content as a successful operation.

Reply:
done